homeassistant/components/evohome.py

```diff
@@ -8,138 +8,153 @@
+                "The server is not contactable. Unable to continue. "
+                "Resolve any errors and restart HA."
+            )
+
+        elif err.response.status_code == HTTP_TOO_MANY_REQUESTS:
+            _LOGGER.error(
+                "setup(): Failed to connect with the vendor's web servers. "
+                "You have exceeded the api rate limit. Unable to continue. "
+                "Wait a while (say 10 minutes) and restart HA."
+            )
+
+        else:
+            raise  # we dont expect/handle any other HTTPErrors
 
-    finally:  # Redact username, password as no longer needed.
+        return False  # unable to continue
+
+    finally:  # Redact username, password as no longer needed
         evo_data['params'][CONF_USERNAME] = 'REDACTED'
         evo_data['params'][CONF_PASSWORD] = 'REDACTED'
 
     evo_data['client'] = client
+    evo_data['status'] = {}
 
-    # Redact any installation data we'll never need.
-    if client.installation_info[0]['locationInfo']['locationId'] != 'REDACTED':
-        for loc in client.installation_info:
-            loc['locationInfo']['streetAddress'] = 'REDACTED'
-            loc['locationInfo']['city'] = 'REDACTED'
-            loc['locationInfo']['locationOwner'] = 'REDACTED'
-            loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+    # Redact any installation data we'll never need
+    for loc in client.installation_info:
+        loc['locationInfo']['locationId'] = 'REDACTED'
+        loc['locationInfo']['locationOwner'] = 'REDACTED'
+        loc['locationInfo']['streetAddress'] = 'REDACTED'
+        loc['locationInfo']['city'] = 'REDACTED'
+        loc[GWS][0]['gatewayInfo'] = 'REDACTED'
 
-    # Pull down the installation configuration.
+    # Pull down the installation configuration
     loc_idx = evo_data['params'][CONF_LOCATION_IDX]
 
     try:
         evo_data['config'] = client.installation_info[loc_idx]
-
     except IndexError:
         _LOGGER.warning(
-            "setup(): Parameter '%s' = %s , is outside its range (0-%s)",
+            "setup(): Parameter '%s'=%s, is outside its range (0-%s)",
             CONF_LOCATION_IDX,
             loc_idx,
             len(client.installation_info) - 1
         )
-
         return False  # unable to continue
 
-    evo_data['status'] = {}
-
     if _LOGGER.isEnabledFor(logging.DEBUG):
         tmp_loc = dict(evo_data['config'])
         tmp_loc['locationInfo']['postcode'] = 'REDACTED'
-        tmp_tcs = tmp_loc[GWS][0][TCS][0]
-        if 'zones' in tmp_tcs:
-            tmp_tcs['zones'] = '...'
-        if 'dhw' in tmp_tcs:
-            tmp_tcs['dhw'] = '...'
+        if 'dhw' in tmp_loc[GWS][0][TCS][0]:  # if this location has DHW...
+            tmp_loc[GWS][0][TCS][0]['dhw'] = '...'
+
+        _LOGGER.debug("setup(): evo_data['config']=%s", tmp_loc)
+
+    hass.async_create_task(
+        async_load_platform(hass, 'climate', DOMAIN, {}, hass_config))
```

Comment:
Call `load_platform` directly.

Reply:
Sorry - re-read & I am not sure if you mean:
```python
hass.async_create_task(load_platform(hass, 'climate', DOMAIN, {}, hass_config))
```
or:
```python
load_platform(hass, 'climate', DOMAIN, {}, hass_config)
```
or even:
```python
async_load_platform(hass, 'climate', DOMAIN, {}, hass_config))
```

Reply:
We're in sync context so have to use the sync api. Call `load_platform` directly, your second example.

Reply:
OK, done - will push now.